File: DuDoong-Api/src/main/java/band/gosrock/DuDoongApiServerApplication.java
package band.gosrock;


import java.util.Arrays;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.context.annotation.Bean;
import org.springframework.core.env.Environment;
import org.springframework.web.filter.ForwardedHeaderFilter;

@RequiredArgsConstructor
@SpringBootApplication
@Slf4j
public class DuDoongApiServerApplication implements ApplicationListener<ApplicationReadyEvent> {

    private final Environment environment;

    public static void main(String[] args) {
        SpringApplication.run(DuDoongApiServerApplication.class, args);
    }

    @Bean
    ForwardedHeaderFilter forwardedHeaderFilter() {
        return new ForwardedHeaderFilter();
    }

    @Override
    public void onApplicationEvent(ApplicationReadyEvent event) {
        log.info("applicationReady status" + Arrays.toString(environment.getActiveProfiles()));
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/DuDoongApiServerApplication.kt
package band.gosrock

import org.slf4j.LoggerFactory
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.ApplicationListener
import org.springframework.context.annotation.Bean
import org.springframework.core.env.Environment
import org.springframework.web.filter.ForwardedHeaderFilter

@SpringBootApplication
class DuDoongApiServerApplication(
    private val environment: Environment
) : ApplicationListener<ApplicationReadyEvent> {

    private val log = LoggerFactory.getLogger(DuDoongApiServerApplication::class.java)

    @Bean
    fun forwardedHeaderFilter(): ForwardedHeaderFilter = ForwardedHeaderFilter()

    override fun onApplicationEvent(event: ApplicationReadyEvent) {
        log.info("applicationReady status${environment.activeProfiles.contentToString()}")
    }
}

fun main(args: Array<String>) {
    SpringApplication.run(DuDoongApiServerApplication::class.java, *args)
}
